AsyncDisplayKit/TextKit/ASTextKitRenderer.mm

```diff
@@ -203,42 +238,50 @@ - (void)drawInContext:(CGContextRef)context bounds:(CGRect)bounds;
     [self _calculateSize];
   }
 
+  bounds = CGRectIntersection(bounds, { .size = _constrainedSize });
   CGRect shadowInsetBounds = [[self shadower] insetRectWithConstrainedRect:bounds];
 
   CGContextSaveGState(context);
   [[self shadower] setShadowInContext:context];
   UIGraphicsPushContext(context);
 
   LOG(@"%@, shadowInsetBounds = %@",self, NSStringFromCGRect(shadowInsetBounds));
-  
-  [[self context] performBlockWithLockedTextKitComponents:^(NSLayoutManager *layoutManager, NSTextStorage *textStorage, NSTextContainer *textContainer) {
-    
-    NSTextStorage *scaledTextStorage = nil;
-    BOOL isScaled = [self isScaled];
+  BOOL isScaled = [self isScaled];
```

Comment:
Use split view here. I had to change the indentation of the slow-path rendering code because early return wasn't an attractive option, since we need to pop the context and restore the state. So `if-else` it is, and GitHub isn't as smart as Phabricator with indentation-only changes.